db/statistics.go

```diff
@@ -1130,11 +1130,26 @@ func markColumnExported(day uint64, column string) error {
 	return nil
 }
 
-func GetValidatorIncomeHistoryChart(validatorIndices []uint64, currency string, lastFinalizedEpoch uint64) ([]*types.ChartDataPoint, error) {
-	incomeHistory, err := GetValidatorIncomeHistory(validatorIndices, 0, 0, lastFinalizedEpoch)
-	if err != nil {
-		return nil, err
+func GetValidatorIncomeHistoryChart(validatorIndices []uint64, currency string, lastFinalizedEpoch uint64, days uint64) ([]*types.ChartDataPoint, error) {
```

Comment:
The amount of code in this function could be reduced a bit while readability could be enhanced:
```
func GetValidatorIncomeHistoryChart(validatorIndices []uint64, currency string, lastFinalizedEpoch uint64, days uint64) ([]*types.ChartDataPoint, error) {
	lowerBound := uint64(0)
	upperBound := uint64(0)

	if days > 0 {
		// lookback amount is limited, calculate bounds
		tNow := time.Now()
		lastDay := time.Date(tNow.Year(), tNow.Month(), tNow.Day(), 0, 0, 0, 0, time.UTC)
		firstDay := lastDay.AddDate(0, -1, 0)
		lowerBound = utils.TimeToDay(uint64(firstDay.Unix()))
		upperBound = utils.TimeToDay(uint64(lastDay.Unix()))
	}

	incomeHistory, err := GetValidatorIncomeHistory(validatorIndices, lowerBound, upperBound, lastFinalizedEpoch)
	if err != nil {
		return nil, err
	}

	var clRewardsSeries = make([]*types.ChartDataPoint, len(incomeHistory))

	for i := 0; i < len(incomeHistory); i++ {
		color := "#7cb5ec"
		if incomeHistory[i].ClRewards < 0 {
			color = "#f7a35c"
		}
		balanceTs := utils.DayToTime(incomeHistory[i].Day)
		clRewardsSeries[i] = &types.ChartDataPoint{X: float64(balanceTs.Unix() * 1000), Y: utils.ExchangeRateForCurrency(currency) * (float64(incomeHistory[i].ClRewards) / 1e9), Color: color}
	}
	return clRewardsSeries, nil // no need to return err here as it is always nil at this point
}
```

This code is UNTESTED but should show you what I mean.